net/dns/resolver/forwarder.go

```diff
@@ -432,3 +501,48 @@ func (p *closePool) Close() error {
 	}
 	return nil
 }
+
+var knownDoH = map[netaddr.IP]string{}
```

Comment:
Rather than baking these into the client, I wonder if it would be better for control to pass along "DNS IP", "DoH IP" and "DoH name". This is probably fine for now, maybe we can transition to that later.

Reply:
Agree. The wire protocol supports this but the plumbing can come later. First I want to see just one part working.